march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -137,30 +143,34 @@ void InertiaEstimator::inertiaEstimate()
   correlationCalculation();
   K_i_ = gainCalculation();
   K_a_ = alphaCalculation();
-  const double torque_e = filtered_joint_torque_[0] - filtered_joint_torque_[1];
-  const double acc_e = filtered_acceleration_array_[0] - filtered_acceleration_array_[1];
+  auto ita = filtered_acceleration_array_.begin();
+  auto itt = filtered_joint_torque_.begin();
+  const double torque_e = *itt - *(++itt);
+  const double acc_e = *ita - *(++ita);
```

Comment:
Is a bit more readable in my opinion and since you don't use it afterwards
```suggestion
  const double torque_e = *itt - *(itt + 1);
  const double acc_e = *ita - *(ita + 1);
```